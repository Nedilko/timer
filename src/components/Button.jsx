import PropTypes from 'prop-types'
import settingsIcon from '../assets/settings.svg'

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  title: PropTypes.string,
  icon:  PropTypes.string
}

function addTitle(title) {
  if (title) {
    return <span className='px-2'>{title}</span>
  }
}

function addIcon(icon) {
  if (icon) {
    return <img className="h-5 w-5 dark:invert" src={icon} alt="Settings"/>
  }
}

function Button({onClick, title, icon}) {
  return (
    <div className="cursor-pointer p-2 border-slate-600 rounded" onClick={onClick}>
      {addIcon(icon)}
      {addTitle(title)}
    </div>
  )
}

export default Button