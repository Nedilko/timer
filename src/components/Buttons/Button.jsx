import PropTypes from 'prop-types'

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  title: PropTypes.string
}

function Button({onClick, title}) {
  return (
    <div className="select-none	cursor-pointer p-2 border-slate-600 rounded" onClick={onClick}>
      <span className='px-2'>{title}</span>
    </div>
  )
}

export default Button