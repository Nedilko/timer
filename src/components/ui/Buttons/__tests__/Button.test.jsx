import Button from "../Button";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Button", () => {
  const clickHandler = jest.fn();
  beforeEach(() => {
    clickHandler.mockClear();
  });
  it("should be in the document", () => {
    render(<Button title="test button" onClick={clickHandler} />);

    expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("should handle click", async () => {
    render(<Button title="test button" onClick={clickHandler} />);

    await userEvent.click(screen.getByRole("button"));

    expect(clickHandler).toHaveBeenCalledTimes(1);
  });
});
