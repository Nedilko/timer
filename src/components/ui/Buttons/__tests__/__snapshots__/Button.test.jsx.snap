// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be in the document 1`] = `
<div
  class="select-none	cursor-pointer p-2 border-slate-600 rounded"
  role="button"
>
  <span
    class="px-2"
  >
    test button
  </span>
</div>
`;
